back/src/main/java/com/baba/back/oauth/service/MemberService.java

```diff
@@ -179,6 +182,56 @@ private Babies saveRelationsAndGetBabies(List<Invitation> invitations, String re
                 .toList());
     }
 
+    public void createGroup(String memberId, CreateGroupRequest request) {
+        final Member member = getMember(memberId);
+        final List<Baby> myBabies = getMyBabies(member);
+        final List<RelationGroup> relationGroups = getRelationGroupsByBabies(myBabies);
+
+        validateRelationGroup(relationGroups, request.getRelationGroup());
+
+        saveRelationGroups(request, myBabies);
+    }
+
+    private List<Baby> getMyBabies(Member member) {
+        final Relations relations = new Relations(relationRepository.findAllByMember(member));
+        final List<RelationGroup> myFamilyGroup = relations.getMyFamilyGroup();
+
+        final List<Baby> babies = myFamilyGroup.stream()
+                .map(RelationGroup::getBaby)
+                .toList();
+
+        if (babies.isEmpty()) {
+            throw new RelationNotFoundException("가족 관계인 아기가 존재하지 않습니다.");
+        }
+
+        return babies;
+    }
+
+    private List<RelationGroup> getRelationGroupsByBabies(List<Baby> babies) {
+        return relationGroupRepository.findAllByBabyIn(babies);
+    }
+
+    private void validateRelationGroup(List<RelationGroup> relationGroups, String relationGroupName) {
+        relationGroups.stream()
+                .collect(Collectors.groupingBy(RelationGroup::getBaby))
+                .forEach((baby, groupsByBaby) -> groupsByBaby.stream()
+                        .filter(relationGroup -> relationGroup.hasEqualName(relationGroupName))
+                        .findAny()
+                        .ifPresent(relationGroup -> {
+                            throw new RelationGroupBadRequestException("이미 존재하는 그룹 이름입니다.");
+                        }));
```

Comment:
```suggestion
        private void validateRelationGroup(List<RelationGroup> relationGroups, String relationGroupName) {
        relationGroups.stream()
                .collect(Collectors.groupingBy(RelationGroup::getBaby))
                .forEach((baby, groupsByBaby) -> {
                    if (groupsByBaby.stream()
                            .anyMatch(relationGroup -> relationGroup.hasEqualName(relationGroupName))) {
                        throw new RelationGroupBadRequestException("이미 존재하는 그룹 이름입니다.");

                    }
                });
    }

```

비슷한 로직이지만, match 되는 것을 발견하면 바로 예외를 터뜨린다는 점에서 조금 낫지 않을까 싶네요.

근데 여기서 의문인게 굳이 baby로 그룹화를 해야할까요?

Reply:
> 비슷한 로직이지만, match 되는 것을 발견하면 바로 예외를 터뜨린다는 점에서 조금 낫지 않을까 싶네요.
좋습니다!!

생각해보니 굳이 baby로 그룹화하지 않아도 되는군요! 변경하겠습니다